indico/modules/users/controllers.py

```diff
@@ -639,3 +639,16 @@ class RHUserSearchInfo(RHProtected):
     def _process(self):
         external_users_available = any(auth.supports_search for auth in multipass.identity_providers.itervalues())
         return jsonify(external_users_available=external_users_available)
+
+
+class RHUserBlock(RHUserBase):
+    def _block(self, blocked):
+        self.user.is_blocked = blocked
+        flash(_('The user has been {}.'.format('blocked' if blocked else 'unblocked')), 'success')
+        return jsonify(success=True)
+
+    def _process_PUT(self):
+        self._block(True)
```

Comment:
I don't think using a separate method is useful considering my i18n comment above. Having 3 lines here is much simpler and more readable:

- set the blocked flag
- flash a message
- (currently missing) use the `logger` to log who's blocking who
- return json (you can use `jsonify_data(flash=False)` to already have the success flag included btw!)

Reply:
I agree, this is over-engineering.